### PR TITLE
Make unanalyzable functions "Angelic"

### PR DESCRIPTION
**Pull request by hermanventer**
## Description

A function that contains calls to other functions for which no body can be obtained and for which no summary has been provided via the standard summary store or the foreign_contracts mechanism cannot be accurately analyzed, leading to many false positives. Right now, the sheer number of messages that are meaningless and hard to make go away are a barrier to adoption.

Consequently "unanalyzable" functions are now marked as "angelic" (which is to say, we simply assume that they are correctly implemented). Any function that calls an angelic function itself becomes angelic, because the lack of post conditions from the angelic call makes the analysis of the caller inaccurate and noisy.

Needless to say, MIRAI now finds very few issues. With time this will improve as more functions without bodies are eliminated either by the top down analysis phase, or by explicit models.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] API change with a documentation update
- [ ] Additional test coverage
- [ ] Code cleanup or just keeping up with the latest Rustc nightly

## How Has This Been Tested?
./validate.sh
export RUSTFLAGS="-Z always_encode_mir"
cargo clean
cargo build
find . -type f -name lib.rs -exec touch {} ;
RUSTC_WRAPPER=mirai cargo build

### CODE REVIEW

File: checker/src/visitors.rs [hunk 1]
```diff
@@ -145,6 +147,30 @@ impl<'analysis, 'compilation, 'tcx, E> Debug for MirVisitor<'analysis, 'compilat
     }
 }
 
+macro_rules! check_for_early_break {
```
Comment (wrwg) [inline checker/src/visitors.rs]: Add a short comment what this and the next one does?

**Comment by hermanventer** [comment]
There is currently no scenario where a flag will be useful, nor do I foresee such a scenario arising in the near future.

**Comment by wrwg** [comment]
> There is currently no scenario where a flag will be useful, nor do I foresee such a scenario arising in the near future.

Here is the scenario I have in mind: you really want to know whether there any functions with missing bodies anywhere transitively in a certain piece of code. With angelic on by default, you can't see it. Worth, you may turn a previously happily verified piece of code into something which is not longer verified accidentally by just changing a simple implementation detail somewhere deep down transitively.

Or do I misunderstand?

**Comment by hermanventer** [comment]
You'll see such functions by turning on logging.

**Pull request merged**